packages/db-authorization/index.js

```diff
@@ -17,11 +17,13 @@ async function auth (app, opts) {
     app.register(require('./lib/webhook'), opts.webhook)
   }
 
+  const disableAdminSecret = !!(opts.jwt || opts.webhook)
+
   const adminSecret = opts.adminSecret
   const roleKey = opts.roleKey || 'X-PLATFORMATIC-ROLE'
   const anonymousRole = opts.anonymousRole || 'anonymous'
   app.addHook('preHandler', async (request) => {
-    if (adminSecret && request.headers['x-platformatic-admin-secret'] === adminSecret) {
+    if (adminSecret && !disableAdminSecret && request.headers['x-platformatic-admin-secret'] === adminSecret) {
```

Comment:
This disables the admin secret support completely. However, I think it's better to just _not_ support user impersonation and setting any other role than `platformatic-admin`. Somebody with the secret could still do everything (he is an admin), but not impersonate users.

Reply:
But if we use an external auth service (JWT or Webhook) why should we use the `adminSecret`?

- Previous behavior: with `secretAdmin` we basically ignore JWT and Webhook, we create a rule for `platformatic-admin` automatically for each entity ([here](https://github.com/platformatic/platformatic/blob/disable-adminSecret-impersonalization-jwt-webhook/packages/db-authorization/index.js#L76)) and we have user impersonation.
- This PR: If we have JWT or Webook we ignore `secretAdmin` (and user impersonation) and we don't create the `platformatic-admin` rule automatically for each entity

If we want to  have an admin in the JWT or Webhook case, we just need to have them assign a `platformatic-admin` role and add the rules accordingly (and we don't need it to call it `platformatic-admin`, BTW). The only "superpower" that we have with `adminSecret` is user impersonation, that we want to disable in this case. 
We might still want to create rules automatically for`platformatic-admin`, though, I don't have a strong opinion on this.